contracts/EVMScriptFactories/TopUpWhitelistedRecipients.sol

```diff
@@ -0,0 +1,153 @@
+import "../interfaces/IEVMScriptFactory.sol";
+
+/// @notice Creates EVMScript to check limits and top up whitelisted recipients addresses
+contract TopUpWhitelistedRecipients is TrustedCaller, IEVMScriptFactory {
+    // -------------
+    // ERRORS
+    // -------------
+    string private constant ERROR_LENGTH_MISMATCH = "LENGTH_MISMATCH";
+    string private constant ERROR_EMPTY_DATA = "EMPTY_DATA";
+    string private constant ERROR_ZERO_AMOUNT = "ZERO_AMOUNT";
+    string private constant ERROR_WHITELISTED_RECEPIENT_NOT_FOUND =
+        "ERROR_WHITELISTED_RECEPIENT_NOT_FOUND";
+    string private constant ERROR_SUM_EXCEEDS_LIMIT = "SUM_EXCEEDS_LIMIT";
+
+    // -------------
+    // VARIABLES
+    // -------------
+
+    /// @notice Address of Aragon's Finance contract
+    IFinance public immutable finance;
+
+    /// @notice Address of payout token
+    address public immutable token;
+
+    /// @notice Address of WhitelistedRecipientsRegistry
+    WhitelistedRecipientsRegistry public immutable whitelistedRecipientsRegistry;
+
+    // -------------
+    // CONSTRUCTOR
+    // -------------
+
+    constructor(
+        address _trustedCaller,
+        address _whitelistedRecipientsRegistry,
+        address _finance,
+        address _token
+    ) TrustedCaller(_trustedCaller) {
+        finance = IFinance(_finance);
+        token = _token;
+        whitelistedRecipientsRegistry = WhitelistedRecipientsRegistry(
+            _whitelistedRecipientsRegistry
+        );
+    }
+
+    // -------------
+    // EXTERNAL METHODS
+    // -------------
+
+    /// @notice Creates EVMScript to top up whitelisted recipients addressees
+    /// @param _creator Address who creates EVMScript
+    /// @param _evmScriptCallData Encoded tuple: (address[] _whitelistedRecipients, uint256[] _amounts) where
+    /// _whitelistedRecipients - addresses of whitelisted recipients to top up
+    /// _amounts - corresponding amount of tokens to transfer
+    function createEVMScript(address _creator, bytes memory _evmScriptCallData)
+        external
+        view
+        override
+        onlyTrustedCaller(_creator)
+        returns (bytes memory)
+    {
+        (
+            address[] memory whitelistedRecipients,
+            uint256[] memory amounts
+        ) = _decodeEVMScriptCallData(_evmScriptCallData);
+        _validateEVMScriptCallData(whitelistedRecipients, amounts);
+
+        bytes[] memory evmScriptsCalldata = new bytes[](whitelistedRecipients.length);
+        uint256 sum = 0;
+        for (uint256 i = 0; i < whitelistedRecipients.length; ++i) {
+            evmScriptsCalldata[i] = abi.encode(
+                token,
+                whitelistedRecipients[i],
+                amounts[i],
+                "Top up whitelisted recipients"
+            );
+            sum += amounts[i];
+        }
+
+        _checkLimit(sum);
+
+        bytes memory _evmScript_checkAndUpdateLimits = EVMScriptCreator.createEVMScript(
```

Comment:
No, we can't for sure.

because we have two values for `_to`: one is `whitelistedRecipientsRegistry`, and the other is `finance`.

There is another overload exist:
```
    /// @notice Encodes multiple calls to different contracts as EVMScript
    function createEVMScript(
        address[] memory _to,
        bytes4[] memory _methodIds,
        bytes[] memory _evmScriptCallData
    )
```
but it has the following requirements inside:

```
        require(_to.length == _methodIds.length, "LENGTH_MISMATCH");
        require(_to.length == _evmScriptCallData.length, "LENGTH_MISMATCH");
```

Thus, we may accomplish the task by introducing the `concatSctipts` helper.

Reply:
We can create an array where first entry is whitelistRecipientRegisry and other n are finance.

Reply:
Make sense :+1: 

Though, still would have been required additional arrays and IDK which variant is more expressive

Reply:
i don't think that filling 3 array is worse than filling one, but clarity can be improved by using structs for that. 
Anyway, getting rid of BytesLib and concat seems a decent win to tolerate it.

Reply:
thank you! Good idea. Changed variant of EVMScript concatenation. Removed BytesLib